.github/workflows/build.yml

```diff
@@ -74,6 +74,20 @@ jobs:
           cppcheck --template=gcc --inline-suppr --std=c++17 --enable=style,unusedFunction -i externals \
                    -i src/plugins/weatherdata/gfs/g2clib -DCPPCHECK --force --suppressions-list=cppcheck.supp \
                    --library=qt.cfg --library=posix.cfg .
+      - name: Run doxygen
+        working-directory: docs
+        run: |
+          sudo apt-get -y install doxygen graphviz
+          doxygen Doxyfile
+        env:
+          DOXY_SRC_ROOT: ..
+      - name: Publish doxygen
+        uses: peaceiris/actions-gh-pages@v3
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        with:
+          deploy_key: ${{ secrets.APIDOCS_DEPLOY_KEY }}
+          external_repository: swift-project/apidocs
+          publish_dir: ./docs/html
```

Comment:
Is there any specific reason why this is published on GitHub Pages through a branch in another repository?
Alternatively, we could use [1], which is used for https://github.com/swift-project/docs. This approach would not require a branch to deploy the site and could be fully handled in the ``pilotclient`` repo.

[1] https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow

Reply:
Good idea, much simpler